explain-integration-testing-patterns.md

```diff
@@ -0,0 +1,331 @@
+
+        await MicrocksContainerEnsemble.StartAsync();
+    }
+
+    protected override void ConfigureWebHost(IWebHostBuilder builder)
+    {
+        base.ConfigureWebHost(builder);
+        
+        var pastryApiEndpoint = MicrocksContainerEnsemble.MicrocksContainer
+            .GetRestMockEndpoint("API Pastries", "0.0.1");
+        builder.UseSetting("PastryApi:BaseUrl", pastryApiEndpoint);
+        
+        var kafkaBootstrap = KafkaContainer.GetBootstrapAddress()
+            .Replace("PLAINTEXT://", "", StringComparison.OrdinalIgnoreCase);
+        builder.UseSetting("Kafka:BootstrapServers", kafkaBootstrap);
+    }
+
+    public async override ValueTask DisposeAsync()
+    {
+        await base.DisposeAsync();
+        await KafkaContainer.DisposeAsync();
+        await MicrocksContainerEnsemble.DisposeAsync();
+    }
+}
+```
+
+#### Step 2: Test Class Implementation
+```csharp
+public class OrderControllerTests : IClassFixture<MicrocksWebApplicationFactory<Program>>
+{
+    private readonly MicrocksWebApplicationFactory<Program> _factory;
+    private readonly ITestOutputHelper _testOutput;
+
+    public OrderControllerTests(
+        MicrocksWebApplicationFactory<Program> factory,
+        ITestOutputHelper testOutput)
+    {
+        _factory = factory;
+        _testOutput = testOutput;
+    }
+
+    [Fact]
+    public async Task CreateOrder_ShouldReturnCreatedOrder()
+    {
+        // This test class has its own containers
+        using var client = _factory.CreateClient();
+        
+        // Test implementation...
+    }
+}
+```
+
+### Pros and Cons
+
+✅ **Advantages:**
+- Complete isolation between test classes
+- Different configurations per test class
+- No shared state issues
+- Parallel test execution per class
+
+❌ **Disadvantages:**
+- Higher resource usage (multiple containers)
+- Slower overall test execution
+- More complex port management
+- Potential for port conflicts if not handled properly
+
+---
+
+## Pattern 2: ICollectionFixture - Shared Containers (Recommended)
+
+### When to use
+- When you want optimal performance and resource usage
+- When test classes can share the same container configuration
+- When you have many test classes
+- When startup time is a concern
+
+### Architecture
+```csharp
+[Collection(SharedTestCollection.Name)]
+public class MyTestClass : BaseIntegrationTest
+{
+    // All test classes share the same factory instance
+    // Single set of containers for all tests
+}
+```
+
+### Key Benefits
+- **Single Container Instance**: One Microcks + one Kafka container for all tests
+- **Performance Optimized**: ~70% faster test execution
+- **Resource Efficient**: Lower memory and CPU usage
+- **Single Port Allocation**: One Kestrel port for the entire test suite
+
+### Implementation Example
+
+#### Step 1: Shared Collection Definition
+```csharp
+[CollectionDefinition(Name)]
+public class SharedTestCollection : ICollectionFixture<MicrocksWebApplicationFactory<Program>>
```

Comment:
Is it the same `MicrocksWebApplicationFactory` that we have in the first strategy? If yes, then my comment on naming still applies ;-)